src/main/java/com/github/black0nion/blackonionbot/commands/bot/AntiSpoilerCommand.java

```diff
@@ -1,47 +1,49 @@
-		if (Utils.handleRights(guild, author, null, Permission.MESSAGE_MANAGE)) return;
-		if (args.length >= 2) {
-			final String type = args[1];
-			if (type.equalsIgnoreCase("replace")) {
-				guild.setAntiSpoilerType(AntiSpoilerType.REPLACE);
-				message.replyEmbeds(EmbedUtils.getSuccessEmbed(author, guild).addField("antispoilerstatuschanged", cmde.getTranslation("antispoileris", new Placeholder("status", cmde.getTranslation("remove"))), false).build()).queue();
-			} else if (type.equalsIgnoreCase("delete")) {
-				guild.setAntiSpoilerType(AntiSpoilerType.DELETE);
-				message.replyEmbeds(EmbedUtils.getSuccessEmbed(author, guild).addField("antispoilerstatuschanged", cmde.getTranslation("antispoileris", new Placeholder("status", cmde.getTranslation("delete"))), false).build()).queue();
-			} else if (type.equalsIgnoreCase("off")) {
-				guild.setAntiSpoilerType(AntiSpoilerType.OFF);
-				message.replyEmbeds(EmbedUtils.getSuccessEmbed(author, guild).addField("antispoilerstatuschanged", cmde.getTranslation("antispoileris", new Placeholder("status", cmde.getTranslation("off"))), false).build()).queue();
-			} else {
-				cmde.sendPleaseUse();
-			}
+	public void execute(@NotNull SlashCommandEvent cmde, @NotNull SlashCommandInteractionEvent e, BlackMember member, BlackUser author, @NotNull BlackGuild guild, TextChannel channel) {
+		var status = e.getOption(STATUS, OptionMapping::getAsString);
+
+		if (status.equalsIgnoreCase("replace")) {
+			guild.setAntiSpoilerType(AntiSpoilerType.REPLACE);
+			e.replyEmbeds(EmbedUtils.getSuccessEmbed(author, guild).addField("antispoilerstatuschanged", cmde.getTranslation("antispoileris", new Placeholder("status", cmde.getTranslation("remove"))), false).build()).queue();
+		} else if (status.equalsIgnoreCase("delete")) {
+			guild.setAntiSpoilerType(AntiSpoilerType.DELETE);
+			e.replyEmbeds(EmbedUtils.getSuccessEmbed(author, guild).addField("antispoilerstatuschanged", cmde.getTranslation("antispoileris", new Placeholder("status", cmde.getTranslation("delete"))), false).build()).queue();
+		} else if (status.equalsIgnoreCase("off")) {
+			guild.setAntiSpoilerType(AntiSpoilerType.OFF);
+			e.replyEmbeds(EmbedUtils.getSuccessEmbed(author, guild).addField("antispoilerstatuschanged", cmde.getTranslation("antispoileris", new Placeholder("status", cmde.getTranslation("off"))), false).build()).queue();
 		} else {
-			message.replyEmbeds(EmbedUtils.getSuccessEmbed(author, guild).addField(LanguageSystem.getTranslation("antispoilerstatus", author, guild).replace("%status%", LanguageSystem.getTranslation(guild.getAntiSpoilerType().name(), author, guild)), LanguageSystem.getTranslation("howtoantispoilertoggle", author, guild).replace("%command%", "``" + CommandEvent.getCommandHelp(guild, this) + "``"), false).build()).queue();
```

Comment:
I'd still keep this, but with the new way of responding.
This way, you can still check the status of the system